### PR TITLE
add explicit destroy method, and option to destroy or just unlock on …

### PR DESCRIPTION
**Pull request by clintropolis**
…unlock

### CODE REVIEW

File: src/zookeeperLock.ts [hunk 1]
```diff
@@ -79,6 +88,11 @@ export class ZookeeperLock {
     };
 
 
+    /**
+     * connect underlying zookeeper client, with optional delay
+     * @param delay
```
Comment (epistemancer) [inline src/zookeeperLock.ts]: should this be `@param [delay]` to mark it as optional?

File: src/zookeeperLock.ts [hunk 2]
```diff
@@ -130,12 +148,24 @@ export class ZookeeperLock {
...
         this.connected = false;
         return this.connect(this.config.spinDelay);
     };
 
+    /**
+     * wait for a lock to become free for a given key and acquire it, with an optional
+     * timeout upon which the lock will fail. if not currently connected to zookeeper,
+     * this will connect, and on timeout, the lock will disconnect from zookeeper
+     * @param key
+     * @param timeout
```
Comment (epistemancer) [inline src/zookeeperLock.ts]: same here

File: src/zookeeperLock.ts [hunk 3]
```diff
@@ -199,7 +229,14 @@ export class ZookeeperLock {
         });
     };
 
-    public unlock = () : Promise<any> => {
+    /**
+     * unlock a lock, removing the key from zookeeper, and disconnecting
+     * the zk client and all event listeners. By default this also destroys
+     * the lock and removes event listeners on the locks 'signals' event
+     * @param destroy - remove listeners from lock.signal in addition to disconnecting zk client on completion, defaults to true
```
Comment (epistemancer) [inline src/zookeeperLock.ts]: `@param [destroy=true]` ?

File: src/zookeeperLock.ts (hunk 1 above)
Comment (clintropolis) [inline src/zookeeperLock.ts]: probably, i don't know jsdoc that well :smile: i'll update

Comment (epistemancer) [inline src/zookeeperLock.ts]: I don't really either but found: http://usejsdoc.org/tags-param.html

**Comment by epistemancer** [comment]
:see_no_evil:

**Pull request merged**